src/account/PluginManagerInternals.sol

```diff
@@ -886,7 +888,20 @@ abstract contract PluginManagerInternals is IPluginManager {
 
         delete _storage.pluginData[oldPlugin];
 
-        // @TODO: Apply Data migration
+        // Retrieve data from the old plugin for migration
+        bytes memory migrationData = IPlugin(oldPlugin).getDataForMigration();
+
+        // Pass the migration data to the new plugin
+        try IPlugin(newPlugin).onReplaceForNewPlugin(migrationData) {} 
+        catch (bytes memory revertReason){
+            revert onReplaceForNewPluginFailed(newPlugin, revertReason);
+        }
+
+        // Call the old plugin's clean-up function
+        try IPlugin(oldPlugin).onReplaceForOldPlugin() {} 
+        catch (bytes memory revertReason){
+            revert onReplaceForOldPluginFailed(oldPlugin, revertReason);
```

Comment:
권한 체크나 유효성 체크 코드가 없는 것 같은데, 필요 없는 것인지 의견 부탁합니다! (저는 맥락을 잘 몰라서)

Reply:
I was thinking of a case where `onReplaceForOldPlugin` works and the plugin data is cleaned up for migration, but `onReplaceForNewPlugin` fails. There could also be a vice versa case, which I think wonj1012 already mentioned. I don't think we could recover the plugin data in this case if the function doesn't revert it. what do you think about this?

Reply:
Maybe there's a difference in the level of security. Ability to uninstall / replace a plugin is really important in the situation that a user installed a malicious plugin, which imo the onUninstall is designed not to revert (I think it's similar to emergency exit or something). The onInstall and onReplaceForNewPlugin reverts when it fails, since it's doing an intial setup necessary for plugins.

Reply:
Understood what you say and edited the code.